checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -300,6 +301,25 @@ public Void visitMemberSelect(MemberSelectTree tree, AnnotatedTypeMirror type) {
         }
 
         /**
+         * Compound assignments aren't desugared correctly in some way, which prevents x from
+         * being @Positive after, e.g., this code is executed: "x = 5; x -= 2;". This code is more
+         * general and ensures that any identifier with a compile-time known value is represented
+         * correctly.
+         */
+        @Override
+        public Void visitIdentifier(IdentifierTree tree, AnnotatedTypeMirror type) {
```

Comment:
Also, the LBC should allow the following code.
````
@IntVal({1, 2}) int getOneOrTwo() {...}
void test() {
  int[] a = new int[getOneOrTwo()];
}
````
Which should work once my above suggestion is implemented.